src/main/java/duke/command/AddCommand.java

```diff
@@ -0,0 +1,57 @@
+package duke.command;
+
+import duke.component.Parser;
+import duke.component.Storage;
+import duke.component.TaskList;
+import duke.component.Ui;
+import duke.task.Task;
+
+/**
+ * Represents a command for adding tasks.
+ */
+public class AddCommand extends Command {
+    /**
+     * Creates a command for adding tasks.
+     * @param input the input command classified as AddCommand
+     */
+    public AddCommand(String input) {
+        super(input);
+    }
+
+    /**
+     * Executes the command, prints the result on ui and writes to source data file.
+     * @param ui the user interface object that is currently running
+     * @param list the current list of tasks
+     * @param storage the storage-writing object that is currently running
+     * @return the string that is to be printed on ui if the adding is successful
+     * @throws InvalidCommandException if the input command doesn't make sense and states why
+     */
+    @Override
+    public String execute(Ui ui, TaskList list, Storage storage) throws InvalidCommandException {
```

Comment:
Perhaps it is better to name your TaskList object as taskList instead of list to avoid confusion and make it clearer?